akka/src/main/scala/chatapp/Chat.scala

```diff
@@ -16,81 +16,79 @@ object Action extends Enumeration {
 import Action._
 
 class BehaviorFactory(compute: Int, post: Int, leave: Int, invite: Int) {
-  def apply(dice: DiceRoll): Action = {
-    val pick = dice()
+  val _compute: Int = compute
+  val _post: Int = _compute + post
+  val _leave: Int = _post + leave
+  val _invite: Int = _leave + invite
```

Comment:
Why is this needed?
The arguments to the constructor are visible in the body of the class for direct use, no?

Reply:
see https://github.com/sblessing/chat-app/blob/ce80afdfea43cad366bb8b0ccd6341be9da22fa8/pony/chat.pony#L42
Yes, the arguments are visible in the body.

Reply:
I don't understand this reply.
In Pony, you got a constructor and need to initialize the fields explicitly.
In Scala, this is not necessary, because the constructor's arguments are always accessible. They are not in Pony.

Introducing these fields seems unnecessary.

Reply:
I added the fields because of the calculations to set them to same value as pony and CAF

Reply:
Ah, my bad. That's my sloppy reading of the code. I only saw the initialization.